testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -6085,19 +6083,278 @@ fn fuzzed_median_fee_rate_estimation_test() {
 
     // Check that:
     // 1) The cost is always the same.
-    // 2) The estimated fee rate only either i) moves up, ii) stays within random noise of where it
-    //    was.
+    // 2) Only grows.
     for i in 1..response_estimated_costs.len() {
         let curr_rate = response_top_fee_rates[i] as f64;
         let last_rate = response_top_fee_rates[i - 1] as f64;
-        // Check that either 1) curr_rate is bigger, or 2) is only smaller by random 11% noise.
-        assert!(curr_rate - last_rate > -1.0 * curr_rate * 0.11);
+        // Check that the rate is growing monotonically.
+        assert!(curr_rate >= last_rate);
 
         // The cost is always the same.
         let curr_cost = response_estimated_costs[i];
         let last_cost = response_estimated_costs[i - 1];
         assert_eq!(curr_cost, last_cost);
     }
 
+    // Check the final value is "near" what was expected.
+    assert!(response_top_fee_rates.last().unwrap() - expected_final_value < 0.1);
+
     channel.stop_chains_coordinator();
 }
+
+/// Test the FuzzedWeightedMedianFeeRate with window size 5 and randomness 0. We increase the
+/// amount paid linearly each time. This estimate should grow faster than with window size 10.
+#[test]
+#[ignore]
+fn fuzzed_median_fee_rate_estimation_test_window5() {
+    fuzzed_median_fee_rate_estimation_test(5, 202680.0992)
+}
+
+/// Test the FuzzedWeightedMedianFeeRate with window size 10 and randomness 0. We increase the
+/// amount paid linearly each time. This estimate should grow faster than with window size 5.
+#[test]
+#[ignore]
+fn fuzzed_median_fee_rate_estimation_test_window10() {
+    fuzzed_median_fee_rate_estimation_test(10, 90080.5496)
+}
+
+#[test]
+#[ignore]
+fn use_latest_tip_integration_test() {
+    // The purpose of this test is to check if setting the query parameter `tip` to `latest` is working
+    // as expected. Multiple endpoints accept this parameter, and in this test, we are using the
+    // GetContractSrc method to test it.
```

Comment:
Why is this new test here? Is this testing some new feature for this PR or is it just adding coverage for something previously not covered?

Reply:
something weird is going on with git/github here because this is in `develop` I'm pretty sure

Reply:
Ah, yeah, you're right.